backend/src/utils/reservations.js

```diff
@@ -0,0 +1,93 @@
+const users = require('./users');
+const courses = require('./courses');
+const events = require('./events');
+const validateReservationRequest = (reservationRequest) => {
+    const userId = reservationRequest.userId;
+    const eventId = reservationRequest.eventId;
+    const courseId = reservationRequest.courseId;
+    const ticketType = reservationRequest.ticketType;
+    const ticketPrice = reservationRequest.ticketPrice;
+    const bookingStatus = reservationRequest.bookingStatus;
+
+    const userIdError = users.validateUserId(userId);
+
+    if (userIdError) {
+        return userIdError;
+    }
+
+    const eventIdError = events.validateEventId(eventId);
+
+    if (eventIdError) {
+        return eventIdError;
+    }
+
+    const courseIdError = courses.validateCourseId(courseId);
+
+    if (courseIdError) {
+        return courseIdError;
+    }
+
+    const ticketTypeError = validateTicketType(ticketType);
+
+    if (ticketTypeError) {
+        return ticketTypeError;
+    }
+
+    const ticketPriceError = validateTicketPrice(ticketPrice);
+
+    if (ticketPriceError) {
+        return ticketPriceError;
+    }
+
+    const bookingStatusError = validateBookingStatus(bookingStatus);
+
+    if (bookingStatusError) {
+        return bookingStatusError;
+    }
+};
+
+const validateBookingStatus = (bookingStatus) => {
+    if (typeof bookingStatus !== 'number' || bookingStatus) {
+        return 'booking status is not valid';
+    }
+};
+
+const validateTicketPrice = (ticketPrice) => {
+    if (typeof ticketPrice !== 'number' || ticketPrice <= 0) {
+        return 'Ticket Price is not valid';
+    }
+};
+
+const validateTicketType = (ticketType) => {
+    if (typeof ticketType !== 'string' || ticketType !== 'Single_Ticket') {
```

Comment:
I suppose there is going to be more ticket types in the future, right? You could already format it to allow them by doing:

```
if (typeof ticketType !== 'string' || !['Single_Ticket'].includes(ticketType)) {
```